ingest-v2/src/main/kotlin/com/microsoft/azure/kusto/ingest/v2/source/BlobSourceInfo.kt

```diff
@@ -90,5 +92,114 @@ class BlobSourceInfo : AbstractSourceInfo {
             blobUploadContainer: BlobUploadContainer,
         ): BlobSourceInfo =
             fromLocalSource(streamSourceInfo, blobUploadContainer)
+
+        // batch convert multiple LocalSource objects to BlobSourceInfo using parallel uploads
+        suspend fun fromLocalSourcesBatch(
+            localSources: List<LocalSource>,
+            blobUploadContainer: BlobUploadContainer,
+        ): BatchConversionResult {
+            if (localSources.isEmpty()) {
+                return BatchConversionResult(emptyList(), emptyList())
+            }
+
+            logger.info(
+                "Starting batch conversion of {} local sources",
+                localSources.size,
+            )
+
+            val uploadSources =
+                localSources.map { source ->
```

Comment:
Looking at compression, we lack a test today

use a simple parquet file from this [URL](https://www.timestored.com/data/sample/parquet), and write a test with Ingest format Parquet

Similarly generate an avro file, also a simple GZIP file and upload it. They should all be double compressed